File: src/polygonconfig.ts
import { Random } from "./random";

export { PolygonConfig, RandomPolygonConfig };

class PolygonConfig {
  constructor(
    private sides: number,
    private circles: boolean,
    private arcs: boolean
  ) {}

  Sides(): number {
    return this.sides;
  }

  Circles(): boolean {
    return this.circles;
  }

  Arcs(): boolean {
    return this.arcs;
  }
}

function RandomPolygonConfig(random: Random): PolygonConfig {
  return new PolygonConfig(
    3 + Math.round(random.nextFloat() * 5),
    random.nextFloat() >= 0.5,
    random.nextFloat() >= 0.5
  );
}

File: UnityExample/Assets/Transmutation/Scripts/PolygonConfig.cs
using UnityEngine;

namespace EliCDavis.Transmutation
{
    public class PolygonConfig
    {
        private int sides;

        private bool circles;


        public PolygonConfig(int sides, bool circles)
        {
            this.sides = sides;
            this.circles = circles;
        }

        public static PolygonConfig RandomConfig()
        {
            return new PolygonConfig(
                Random.Range(3, 8),
                Random.value > .5
            );
        }

        public int Sides(){
            return sides;
        }

        public bool Circles(){
            return circles;
        }

    }

}